py4j-java/src/main/java/py4j/GatewayServer.java

```diff
@@ -427,6 +427,35 @@ public GatewayServer(Object entryPoint, int port, InetAddress address, int conne
 		this.sSocketFactory = sSocketFactory;
 	}
 
+	/*
+	 * @param gateway gateway instance (or subclass).  Must not be <code>null</code>.
+	 * @param port the host port to usu
+	 * @param address the host address to use
+	 * @param connectTimeout the connect timeout (ms)
+	 * @param readTimeout the read timeout (ms)
+	 * @param customCommands any customCommands to use.  May be <code>null</code>
+	 * @param sSocketFactory socketFactory to use.  Must not be <code>null</code>
+	 */
+	public GatewayServer(Gateway gateway, int port, InetAddress address, int connectTimeout, int readTimeout,
```

Comment:
Hi Barthelemy,

I would be OK with this, although I think it could be confusing having a Gateway parameter to GatewayServerBuilder...since Gateway incorporates a number of others.  But your call.  As long as it's possible to define a subclass of Gateway and have it used for GatewayServer.   It's a little unfortunate that GatewayServer already has many params, and since it's API you can't really change that at this point without breaking many clients/API.